Added alternative crafting options for welding goggles

Cosmitz opened the pull request:
Welding goggles are the cornerstone of starting to craft vehicles as well as a variety of parts, and often enough the entire progres is stalled by not having a pair of welding goggles, far past making an improvised welder at least. And then in turn, the welding goggles are stalled up by finding a single spray painter.

I reworked the crafting recipe to exclude the spray painter, but including half the charges of any of the paint cans (minus white), charcoal or coal, all which are more accessibily findable if you aim to look for them specifically.

angoddu commented:
Why not allow spray cans as well as the other options?

Cosmitz commented:
Because the spray can itself is a tool, not a consumable. Think of it more like an air compressor spray that you attach a paint can to.

angoddu commented:
> This is a spray can, filled with paint.  Use this tool to make graffiti on the floor.

But it is a can of spray paint, not air. The charges in the can represent the amount of paint in the can

angoddu commented:
Yes a 5 charge consume would be great

Cosmitz commented:
Nope can't add it as a charge, but added as a straight one unit use.

angoddu commented:
The one charge seems to be a bit unbalanced with the other options. I liked the idea of requiring 5 charges

Cosmitz commented:
Sadly, not a possibility, but the option of using it is there.

angoddu commented:
What about making a second recipe for the with paint/coal recipe and leave the existing one there with the tool charge set to 5 instead of 1

Cosmitz commented:
I'd imagine it's a bit frowned upon for creating additional recipies for the same item given i don't think we have a precedent. Also given the narrow use of spray cans anyway it's either using them for the goggles, using them for 'writing' on floors which i'm not even sure it works anymore, and for the flamethrower recipe, there isn't much competition.

pisskop commented:
using globs of paint to create a transparent eye protection apparatus seems more counter intuitive and maybe bright-nights-ey than an acceptable coregame recipe.

Cosmitz commented:
Neither is layering paint on via a spraycan in order to create a shade 7.0 required for oxy torch cutting. But to counter that specific quirk, you can thin paints. In any case i feel it's a reasonable recipe aimed more at the context of the game as is, providing a gameplay solution to an existing bottleneck.

kevingranade commented:
I'm not really a fan of the spraypaint recipe in the first place, I kind of suspect someone took this: http://www.doityourself.com/stry/how-to-build-your-own-welding-goggles and didn't actually read the directions to find out that the spraypaint is only used for making the rest of the goggles opaque, not for the lenses.

As for shitty improvised welding goggles, the only things I know of that would somewhat work are sunglass lenses, possibly several layers of them since you're frequently dealing with high intensity UV.

Can we maybe just have sunglasses spawn randomly and fairly frequently and make those satisfy the lens requirement?  That sounds like something that could reasonably be common enough to meet your goal.  Zombies can wear them, they can appear in convenience stores and at the front of a lot of shops, they can spawn in houses and cars.

If you can find some reasonable references to other DIY options, feel free (even if you can find something referencing the paint or spraint concept), but I'm not interested in making these recipes make even less sense than they do now.

Cosmitz commented:
@kevingranade Sunglasses are decently findable and we can increase their spawn a tiny bit, and i can also add another pair (or maybe two pairs?) of sunglasses in the second part of the recipe to satisfy the condition instead of paint. The ideea has been tossed about and it's not a bad one(though there's more light than just UV coming out welding). 

Either way, the final recipe would be something that could be crafted out of 2-3 pairs of sunglasses and some duct tape, if you don't have access to paint/charcoal.

IRL welding is based on a standard called 'shades' of how much light (let's say light) they allow in, with the darker tinted, the safer/better it is, but indeed, some types of welding only require a lower shade rating. Will do a new commit in a bit.

Coolthulhu commented:
I don't see anyone objecting to the change to gameplay, but there is a minor technical problem:
You forgot one space, meaning the recipe is not in correct format (it has to be exactly identical to what would linter produce).
To ensure your files are always in correct format, run them through this:
http://dev.narc.ro/cataclysm/format.html

Cosmitz commented:
I could have sworn i did. Shrug. Fixed.

kevingranade commented:
I wasn't suggesting sunglasses as an alternative to the paint, I was suggesting it as the ONLY option (unless someone knows of something else that will actually work).

angoddu commented:
While on this topic, can stylish sunglasses be allowed in the recipe? Seems odd that they aren't there currently

Cosmitz commented:
@kevingranade 

I'm one to choose gameplay over ultrarealism, since if we're going full real, there can be no custom-crafted welding goggles. Welding goggles/shields all come with filters that filter out 100% of UV and IR light and you can't just do that by stacking on comercial grade sunglasses or any kind of paint. The addition of the shade rating is literally how much of the visible light spectrum they dim as so the welder can work in confort as well as reduce the maximum amount of brightness that their eyes are subjected to.

But for pure, raw, gameplay considerations given what we have in the game and the certain level of 'makes sense' we allow in it, using paint to dim welding goggles or stackin three pairs of sunglasses is fine enough.

I also have added stylish sunglasses to the recipe.

angoddu commented:
> I also have added stylish sunglasses to the recipe.

Thank you

pisskop commented:
I demand stylish welding googles nao :b

kevingranade commented:
> I'm one to choose gameplay over ultrarealism,

This isn't ultrarealism, this is discarding a nonsense crafting option *now that we have a better alternative*.  There are easy to find sources (including the one I linked above) indicating that people use sunglasses when welding (yes I'm sure it's a terrible idea, but at some point you accept "close enough"), but im not aware of any such indication that people spray or dip lenses in paint for this purpose.

> Plus, the core issue is that they really are a rather silly but poignant bottleneck in progression.

Does replacing spray paint with sunglasses fix the bottleneck or not? "Dip lenses in paint" and whatever is supposed to happen with charcoal aren't options, so we're just comparing spray paint and sunglass lenses here.

Cosmitz commented:
They fix the bottleneck due to a greater availibility of crafting requirements. It works as a gameplay solution for the game as-is without re-examining why the item exists to be crafted in the first place. 

Since if we were to do that:

Spray paint does nothing, it's just aersolised thinner paint, same as layering a thinner coat of existing paint and charcoal/coal can be ground up and made a paste which would fare just as well. In short, you can't make proper modern welding goggles without proper anti UV/IR solutions. If[ this ](http://www.doityourself.com/stry/how-to-build-your-own-welding-goggles)was the original idea for 'spraypaint' goggles, they missed out the part where you have to apply the shade 5 UV tint, which is literally a film of UV blocking transparent material, the same that can be used for cars and something very specific that we wouldn't add just for the sake of crafting one item.

As for ultrarealism: I have stick welded before with no protection. Eyes[ gone red and i started tearing up after a short welding session](https://en.wikipedia.org/wiki/Photokeratitis) while working over at my granparent's house on their balcony. Pleasent? Nope. Life threathening? Nope. Smart? Eehh. Doable? Totally.

So yes, it would be neat to create an underlying system that dealt with eye-burn due to welding unprotected, with conditional debuffs all through CDDA's fantastic trait/disease system, and varied qualities of welding glasses and solutions which can be found which would be scaled to the type of welding instrument you are using, stick/jet/plasma/oxy, as they are not the same in regards to the dangers they pose.

Maybe i or someone else may be inclined in the future to go deeper into welding, as vehicle construction is a core part of CDDA, working to bring the mechanic to the proeminent front which it deserves and allowing more granularity and the option to create gameplay upon the system.

But for this exact moment in time and space, i think this is just a proper fix in the right direction for the CDDA of today that deals with the state of the game as it is now.

angoddu commented:
From what I can tell @Cosmitz wants to allow the use of paint cans in the recipe to make it easier to craft welding goggles without needing to find a spray can. The issue that @kevingranade brings up is that instead of making an unrealistic crafting recipe more unrealistic to change it to be something more realistic by using sunglasses while still increasing the availability of the crafting components, since sunglasses are more common that spray cans anyway. @Cosmitz also proposed to increase the spawn rate of sunglasses to make it even easier to gather enough components for this recipe.

Instead of debating the higher level reasoning as to why/when the change should be made. We should simply address the issue that opened this PR, welding goggles being too hard to craft early game. If we make the recipe only require the base glasses and 2 sunglasses, while also increasing the spawn rate of sunglasses, should be a good compromise on both sides right? Welding goggles are easier to craft and are more realistic.

I know I'm still new here so I may be missing something, but this seems like a lot of debate over fixing the simple issue of needing welding goggles easier to get. Hope this helps :)

Cosmitz commented:
I think @kevingranade means that if we are going to touch this, we should rework it to be entirely more 'down to earth' than what it is currently, and thought the 'stacking of sunglasses' to be an option to do that, which i mentioned that it really isn't. I countered with the fact that a proper grounding of this feature is a larger body of work waiting to happen, not bound to just fixing one recipe, and for this exact moment, the exact issue this PR yields to fix would be fixed by it under the current status of the game and its current internal logic.

angoddu commented:
That doesn't mean we need to completely rework the welding system in one go. We could do it in pieces with the sunglasses change being a good first step in that direction.

Cosmitz commented:
A viable option if we want to allow welding goggles to be made cheaply and still throw them around without making it ridiculous to find welding goggles in a kitchen, we can create a new item, eclipse glasses. Glasses for viewing solar eclipses actually go higher than welding goggles/lenses in terms of protection since they're made for looking exactly straight at the sun. Sunglasses are only meant for background UV and often feature no IR filters.

If you want a link, here's one from NASA: https://eclipse2017.nasa.gov/safety

"Viewing with Protection -- Experts suggests that one widely available filter for safe solar viewing is welders glass of sufficiently high number. The only ones that are safe for direct viewing of the Sun with your eyes are those of Shade 12 or higher. These are much darker than the filters used for most kinds of welding. If you have an old welder's helmet around the house and are thinking of using it to view the Sun, make sure you know the filter's shade number. If it's less than 12 (and it probably is), don't even think about using it to look at the Sun. Many people find the Sun too bright even in a Shade 12 filter, and some find the Sun too dim in a Shade 14 filter — but Shade 13 filters are uncommon and can be hard to find. "

So yeah, we can litter eclipse glasses to be used in the recipe.

kevingranade commented:
As @angoddu points out, I'm not asking for some kind of overhaul, I'm asking for:
* Add sunglasses to recipe.
* Remove all paint from recipe.
* (optional) bump up sunglasses spawn rates.

We agree that stacked sunglasses are a pretty shitty approximation of welding goggles, and we ALSO agree that the game deeply needs some kind of improvised welding goggles.

My point is that painting lenses isn't just a bad approximation, its utter nonsense.

I'd love for a better option, but I don't have one, and sunglass lenses is the best we have.

kevingranade commented:
Lol eclipse glasses, that's fine, but they're even rarer than welding goggles so it doesn't really help :D

angoddu commented:
It's actually funny that you mention eclipse glasses because I am about to purchase a pair of them to view the eclipse in a few days :P But we could add that as a rare item but it still doesn't solve the issue of getting welding goggles early on... Maybe those could be fancy welding goggles :D

Cosmitz commented:
I'm pretty sure everyone in the USA will have a pair of eclipse glasses in some corner or on some shelf for entire years to come after this year. In my country we had perfect front-row seats to the eclipse in 1999 and i still have a pair somewhere in a closet. :) 

This recipe would entirely replace what we currently use for it, but i am going to sprinkle them quite generously, as well as rare drops from zombies.

Ok will have a first pass in a few.

angoddu commented:
I don't like that the welding goggles now can only be crafted with the eclipse glasses. That makes them even harder to get in general, not just early on

Cosmitz commented:
Just commited them to houses/cars/offices, and that's alongside all the current normal locations for welding goggles.

There are more places that spawn them now than there were places that spawned the spray can, just as an idea.

kevingranade commented:
You've made eclipse glasses ludicrously common, please reign it in, a lot.
Also there's a ton of styling issues, probably tabs.

Cosmitz commented:
Ran it through http://dev.narc.ro/cataclysm/format.html and it outputted that so i would assume it wasn't pre-formatted in that styling system. If there is something else i should use instead, let me know.

Removed half of the more special locations as spawns, lowered chances overall down to rarer items, removed clothing spawn.

BevapDin commented on data/json/items/armor.json:
```diff
@@ -2909,6 +2909,28 @@
     "flags": [ "WATER_FRIENDLY", "SUN_GLASSES" ]
   },
   {
+    "id": "eclise_glasses",
```
There is a `p` missing. The game should have complained that the new recipe uses the undefined item "eclipse_glasses".

Cosmitz replied:
Fixed that in a later commit.

kevingranade commented on data/json/items/armor.json:
```diff
@@ -2909,6 +2909,28 @@
+    "type": "ARMOR",
+    "name": "pair of eclipse glasses",
+    "name_plural": "pairs of eclipse glasses",
+    "description": "After the last eclipse before the Cataclysm stirred up huge interest in both social media as well as in traditional publication, these glasses started appearing everywhere. After the grand event passed pairs of these glasses ended up in many a lost shelf or drawer to be forgotten.",
```
This description is the story of why you're adding them to the game, not the description of the item itself.  Please just describe the item.

kevingranade commented:
This went totally off the rails:
1. Restore the sunglasses as a welding mask option.
2. Eclipse glasses don't rate more than a "1" entry in any of the places you have them spawning, and they shouldn't spawn in the possession of any monsters or NPCs.  The current settings have them being unbelievably common, jarringly so.
3. There shouldn't be non-meaningful changes, you ended up shifting around formatting when you added and then removed things in several places.

Cosmitz commented:
1) If we're following the realism angle, sunglasses don't make a lick of sense, no more than painting, so i don't see a reason to include them.

2) Atm eclipse glasses are not spawning in any monsters or NPCs, all are static locations. However, the whole point of the PR were for welding goggles to be /easier/ to craft, not harder. 

3) I'll rework the stylised document to correspond to the original but after working for a bit in different jsons over the last week, i noticed a distinct lack of congruent style between them and even a lack of stylisation between sections in the same json. Either way, agreed, not the PR to start changing that.

kevingranade commented:
1. Crafting welding goggles with sunglass lenses is *still* the least-bad compromise we have. Regardless of how common eclipse glasses are at some specific point in time, they arent generally common, and worse theyre extremely obscure.
2. That's why I said to keep sunglasses in the recipe.

Cosmitz commented:
Readded the variety of sunglasses as optional craftables.

angoddu commented:
Why was the duct tape/medical tape amounts doubled?

Cosmitz commented:
To offset the fewer diversity overall of requirements needed. Plus it's still within how much you can find on a single pick-up.

Cyrano7 commented:
Its looks like the only purpose and the only future purpose of the eclipse glasses is to craft the welding goggles. So as soon as you have a pair of welding goggles they are just clutter. I don't think having another such narrowly used item is good. Might be just better to keep the recipe with the sunglasses for now until some other form of advanced crafting comes about.

Cosmitz commented:
I tried to make them more than just clutter. They also give blindness in normal light so that makes for a cheap blindfold alternative, as well as having Glare Reduction 3 for whatever else it may be used as a quality in the game. It's also good as a future crafting item since it features lenses/materials which would realistically be useful in other recipes, and are highly specialized.

In any case, the only reason i added said eclipse glasses was because it was stated that we needed a more realistic option if we were going to touch the recipe. This is it. We then re-added normal sunglasses as a compromise to overcome a predisposition of not wanting to spawn more of said eclipse glasses. That's where we are now.

Again, the raw purpose of this PR was to make the welding goggles more easily craftable. My initial option (paint/charcoal/coal) was as fine a solution as just allowing sunglasses, from a realism point of view, and fulfilled the gameplay criterias. The ultrarealistic+semi-relistic option in the current latest PR is again, completely fine for allowing what we wanted initially. If we remove the eclipse glasses, we may need to go and tweak sunglasses drop rates a bit, but in the end, this would be the third or forth refactoring of this PR.

Cyrano7 commented:
@Cosmitz Making them completely blind you is not good ether. Eclipse glasses don't block out all light like a blindfold. The highest glare protection level on a item in the game right now is 2, so having 3 is pointless. As for materials/lenses, almost all eclipse glasses you see are the disposable kind so i'm not sure how useful those would be.

It might be better to just change the recipe to take sunglasses instead for now. Finding 3 pairs of sunglasses is easier then finding the spray can i'm pretty sure. If you think the welding goggles are still to hard to craft/find you can always bump the spawns for the glasses or goggles later.

kevingranade commented:
> I don't think having another such narrowly used item is good.

Project norm is to be inclusive of obscure items, but keep an eye on how common they are.  Also as @Cosmitz points out, they're potentially quite useful in other areas as a crafting material.

The blindness thing is pretty iffy, they're quite bad at covering your entire field of view.  That's minor enough that I'd just patch it out in a merge or just leave it for correction later though, considering that this IS a very nice set of changes that is in high demand.

> The highest glare protection level on a item in the game right now is 2, so having 3 is pointless.

~Build it and they will come~

> As for materials/lenses, almost all eclipse glasses you see are the disposable kind so i'm not sure how useful those would be.

Mine are *quite* fragile, surprisingly so.  This further convinces me that they aren't going to be at all common a month or so after the most recent eclipse, but I think their fragility can be overlooked, if it's really bothering you I'd happily accept a recipe that adds additional lenses to protect the filters.

I think the sunglasses change is the significant one and the eclipse glasses are adding flavor, but that's fine.

Cyrano7 commented:
@kevingranade Build it and they will come? That's kinda putting the cart before the horse huh? Nothing requires glare protection 3 as of yet. Not that that can't change in the future but its not here yet so better to put 2.

Cosmitz commented:
It's not hurting or messing up anything right now by adding it. Is it really a point of contention? I just thought it would be _neat_ even if only for the flavor of it. Tool qualities are all over the place right now with no real use of the gradient of them anyway, oftentimes it's SCREW : 1 and that's it. 

I have no issue dropping it to 2 since that's not what this PR is aiming to push as a change, but 'don't put it in since we don't need it', is not something that should apply to the open-source project of CDDA where i just found yesterday a full anthropomorphic dinosaur suit that someone added at some point. Did we need a dinosaur suit? Of course not. Is it _neat_? Yes it is.

BorkBorkGoesTheCode commented:
Brainstorm: Mylar:
https://en.wikipedia.org/wiki/BoPET#Solar.2C_marine_and_aviation_applications

Cosmitz commented:
Mylar is pretty cool but sadly atm inexisted in CDDA, and we'd need to really entrench it as a crafting material otherwise it'd be an edge-case. Plus for exactly this use, it still needs to be aluminized/coated with a film of aluminum; making welding gogles out of an emergency blanket and aluminum powder kind of goes agains the point of making them more accessible. :)

angoddu commented:
I say we accept this PR as is (potentially without blindness) and open an issue to discuss making a more realistic welding system. That way this PR doesn't get even more crowded than it already is

> i found an anthropomorphic dinosaur suit

P.S. I love using those in the winter as pajamas. Keeps me warm and makes me laugh

Night-Pryanik commented:
> Mylar is pretty cool but sadly atm inexistent in CDDA, and we'd need to really entrench it as a crafting material otherwise it'd be an edge-case. Plus for exactly this use, it still needs to be aluminized/coated with a film of aluminum; making welding gogles out of an emergency blanket and aluminum powder kind of goes agains the point of making them more accessible. :)

We do have aluminized mylar in CDDA in form of `emergency blanket` despite its material is stated as `plastic`. https://en.wikipedia.org/wiki/Space_blanket
Maybe we can salvage something from here?

Cosmitz commented:
I could add it as a component instead of the glasses for what it's worth, however i feel it may be somewhat of an obtuse requirement for the general player to conceptualise.

EIther way, if @kevingranade 's fine with it, i'll add it in. It does go towards the goal of this PR.

kevingranade commented:
Let's just call this one now, though there are a few tweaks I'm going to submit as my own PR.

Cyrano7 commented:
@Cosmitz Coming back to the party late it seems. Yes we actually needed the dino suit in the game. This has nothing to do with whether we need just for fun content. Its just making sure everything is worked out before merge. Sure irl you could find a pair of eclipse glasses. Is it neat? No it is not.

tinukedaya commented:
Coming in late on this, I'm sorry for that, but the question I have is:
    Why would I EVER want to build welding goggles using the eclipse glasses once I find those? Seeing as they have the Glare Protection 3, they are more that suited for welding in game, without any other hassle like getting proper skills, book or other components...

Treah commented:
I hate this change so much :(. Somone said that sunglasses are easy to find... Yeah 22 days of searching and not a single one.... not easy to find if your on a low spawn number. This stalls crafting greatly as finding spray cans is oddly WAY simpler.. Most people never craft them from sunglasses and just look for PBA masks. I feel this needs a rework to not make using the makeshift welder stall for SO SO long.

Night-Pryanik commented:
There are tons of sunglasses dropping from the zeds.

Treah commented:
Yeah not from what I see... This would have been a lot better if you could disassemble the glasses into dark lenses that you then could use to craft them. As it stands you have to find 3 of the exact same kind. Even with normal spawn people still complain about this...

Night-Pryanik commented:
@Treah created a PR (22160) based on your proposal.
